android/app/src/main/java/co/nimblehq/smsforwarder/ui/screens/filter/AllFiltersViewModel.kt

```diff
@@ -1,86 +1,92 @@
 package co.nimblehq.smsforwarder.ui.screens.filter
 
 import androidx.hilt.lifecycle.ViewModelInject
-import co.nimblehq.smsforwarder.domain.data.Filter
 import co.nimblehq.smsforwarder.domain.data.IncomingSmsEntity
 import co.nimblehq.smsforwarder.domain.usecase.ForwardIncomingSmsUseCase
+import co.nimblehq.smsforwarder.domain.usecase.GetFiltersUseCase
 import co.nimblehq.smsforwarder.domain.usecase.ObserveIncomingSmsUseCase
+import co.nimblehq.smsforwarder.domain.usecase.ObserveFiltersUseCase
 import co.nimblehq.smsforwarder.ui.base.BaseViewModel
 import co.nimblehq.smsforwarder.ui.base.NavigationEvent
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.subscribeBy
 import io.reactivex.subjects.BehaviorSubject
 
-interface AllFiltersViewModel {
-    fun navigateToFilterManager()
-}
-
 interface Input {
     fun refresh()
 }
 
-class AllFiltersViewModelImpl @ViewModelInject constructor(
+class AllFiltersViewModel @ViewModelInject constructor(
     private val observeIncomingSmsUseCase: ObserveIncomingSmsUseCase,
-    private val forwardIncomingSmsUseCase: ForwardIncomingSmsUseCase
-) : BaseViewModel(), AllFiltersViewModel, Input {
+    private val forwardIncomingSmsUseCase: ForwardIncomingSmsUseCase,
+    private val observeFiltersUseCase: ObserveFiltersUseCase,
+    private val getFiltersUseCase: GetFiltersUseCase
+) : BaseViewModel(), Input {
 
     val input: Input = this
 
-    private val _data = BehaviorSubject.create<List<Filter>>()
-    val data: Observable<List<Filter>>
+    private val _data = BehaviorSubject.create<List<FilterUiModel>>()
+    val data: Observable<List<FilterUiModel>>
         get() = _data
 
     init {
+        observeFilters()
+        getFilters()
```

Comment:
Do we need to observe and get at the same time? 🤔 
Both of them do get a list from FiltersPersistence. Can you clarify this a bit? 🙏🏻

Reply:
@lydiasama`observeFilters()` means you will observe for the update from Filtering list, the `getFilters()` will get the update and submit the the observable filtering list 😄  Does it make sense to you?

Reply:
Should we use the [navigationResult](https://developer.android.com/guide/navigation/navigation-programmatic) instead of getFilters()? We do update the filter list when we submit at the FilterManagerFragment, right?